TBTK/calc/TightBindingLib/src/FileWriter.cpp

```diff
@@ -1146,4 +1146,97 @@ bool FileWriter::exists(){
 	return exists;
 }
 
+void FileWriter::writeParameterSet(
+    const Util::ParameterSet *parameterSet,
+    std::string name,
+    std::string path
+){
+	init();
+
+	const int ATTRIBUTES_RANK = 1;
+	const hsize_t limits_dims[1] = {1};
+
+	try{
+		stringstream ss;
+		ss << path;
+		if(path.back() != '/')
+			ss << "/";
+		ss << name;
+
+		Exception::dontPrint();
+		H5File file(filename, H5F_ACC_RDWR);
+
+		DataSpace dataspace = DataSpace(ATTRIBUTES_RANK, limits_dims);
+		DataSet dataset = DataSet(file.createDataSet(name + "Int", PredType::STD_I64BE, dataspace));
+
+		for(int n = 0; n < parameterSet->getNumInt(); n++){
+			Attribute attribute = dataset.createAttribute(parameterSet->getIntName(n), PredType::STD_I64BE, dataspace);
+			int value = parameterSet->getIntValue(n);
+			attribute.write(PredType::NATIVE_INT, &value);
+		}
+
+		dataset = DataSet(file.createDataSet(name + "Double", PredType::IEEE_F64BE, dataspace));
+
+        for(int n = 0; n < parameterSet->getNumDouble(); n++){
+			Attribute attribute = dataset.createAttribute(parameterSet->getDoubleName(n), PredType::IEEE_F64BE, dataspace);
+			double value = parameterSet->getDoubleValue(n);
+			attribute.write(PredType::NATIVE_DOUBLE, &value);
+		}
+
+		hsize_t dimension = 2;
```

Comment:
Use
const int COMPLEX_RANK = 1;
int complex_dims[COMPLEX_RANK] = {2};
ArrayType complexDataType(PredType::NATIVE_DOUBLE, COMPLEX_RANK, complex_dims);
(Possible complexType, according to choice in FileWriter)